### ISSUE TITLE
MPI_Win_create errors out when CUDA is enabled

### ISSUE THREAD

**Issue by AboorvaDevarajan** [open-mpi/ompi#11255]
### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
`main` and `v5.0.x`

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
`source`

### If you are building/installing from a git clone, please copy-n-paste the output from `git submodule status`.
```
$ git submodule status
+e770a3362fcea61778b85b4c7cfb7044443c9490 3rd-party/openpmix (v4.2.2-9-ge770a336)
+4262efb88da9292fbf5a29665dcc98eed49c751c 3rd-party/prrte (v3.0.0-2-g4262efb88d)
```

### Please describe the system on which you are running

* Operating system/version: Red Hat Enterprise Linux release 8.6 (Ootpa)
* Computer hardware: ppc64le
* Network type: -

-----------------------------

## Details of the problem

Here is a simple test that recreates this issue:
https://github.com/AboorvaDevarajan/mpi-snippets/blob/main/win_create.c

This fails when CUDA is configured with OMPI:

```
$ mpicc win_create.c -o win_create

$ mpirun -np 2  --mca coll basic,inter,libnbc,cuda ./win_create
[n02:00000] *** An error occurred in MPI_Win_create
[n02:00000] *** reported by process [2310995969,1]
[n02:00000] *** on communicator MPI_COMM_WORLD
[n02:00000] *** MPI_ERR_WIN: invalid window
[n02:00000] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[n02:00000] ***    and MPI will try to terminate your MPI job as well)
```

This test passes when `cuda` is not force selected for the `coll` component.

**Comment by AboorvaDevarajan** [comment]
Upon further debugging was able figure out that the non-zero return code from `mca_coll_cuda_check_buf` when buffer is `NULL` which is introduced recently is causing this issue. 

This should fix the issue: https://github.com/open-mpi/ompi/pull/11256

**Comment by AboorvaDevarajan** [comment]
Upon further debugging it turns out that the non-zero return code from `mca_coll_cuda_check_buf` when buffer is `NULL` which is introduced recently is causing this issue. 

This is a probable fix: https://github.com/open-mpi/ompi/pull/11256

**Comment by janjust** [comment]
closed with #11256
